### PR TITLE
Adds some support and tests for `HydrostaticModel`s

### PR DESCRIPTION
**Pull request by tomchor**
For now only a few functions are tests for hydrostatic models, but it's a start

### CODE REVIEW

**Comment by tomchor** [comment]
All tests pass locally. Not sure the current error means or how to fix it...

File: test/runtests.jl [hunk 1]
```diff
@@ -171,22 +146,32 @@ function test_tracer_diagnostics(model)
...
+                                  closure = scalar_diff),
+              HydrostaticFreeSurfaceModel(; grid, model_kwargs...,
+                                          closure = scalar_diff))
 
-    @info "Testing velocity-only diagnostics"
-    test_vel_only_diagnostics(model)
+    for model in models
+        model_type = split(summary(model), "{")[1]
+        @info "Testing velocity-only diagnostics in $model_type"
+        #test_vel_only_diagnostics(model)
```
Comment (whitleyv) [inline test/runtests.jl:165]: Do you want these all commented out? Or was this just for your own testing of each element...

Comment (tomchor) [reply test/runtests.jl]: Ah, thanks for catching that. Forgot to uncomment these lines :)

**Comment by whitleyv** [comment]
These weren't part of your changes for the review, but do these need to be `YPressure...` / `ZPressure...`?
https://github.com/tomchor/Oceanostics.jl/blob/6a0871b4450c598db0c6ac323248b12fd024d30f/test/runtests.jl#L113
https://github.com/tomchor/Oceanostics.jl/blob/6a0871b4450c598db0c6ac323248b12fd024d30f/test/runtests.jl#L116

**Comment by tomchor** [comment]
> These weren't part of your changes for the review, but do these need to be `YPressure...` / `ZPressure...`?
> 
> https://github.com/tomchor/Oceanostics.jl/blob/6a0871b4450c598db0c6ac323248b12fd024d30f/test/runtests.jl#L113
> 
> 
> https://github.com/tomchor/Oceanostics.jl/blob/6a0871b4450c598db0c6ac323248b12fd024d30f/test/runtests.jl#L116

Not really. These were and still are passing in previous tests so they should be good :)

**Review by whitleyv — APPROVED** [review]

**Pull request merged**